kernel/comps/keyboard/src/i8042_chip/mouse.rs

```diff
@@ -0,0 +1,472 @@
+
+        // Add mouse buttons.
+        capability.set_supported_key(KeyCode::BtnLeft);
+        capability.set_supported_key(KeyCode::BtnRight);
+        capability.set_supported_key(KeyCode::BtnMiddle);
+
+        // Add relative axes for movement.
+        capability.set_supported_relative_axis(RelCode::X);
+        capability.set_supported_relative_axis(RelCode::Y);
+        capability.set_supported_relative_axis(RelCode::Wheel);
+
+        Self {
+            // Standard name for i8042 PS/2 mouse devices.
+            name: "i8042_mouse".to_string(),
+
+            // Physical path describing the device's connection topology
+            // isa0060: ISA bus port 0x60 (i8042 data port)
+            // serio1: Serial I/O device 1 (second PS/2 port)
+            // input0: Input device 0 (first mouse device on this controller)
+            phys: "isa0060/serio1/input0".to_string(),
+
+            // Unique identifier - empty string because traditional i8042 mice
+            // don't have unique hardware identifiers
+            uniq: "".to_string(),
+
+            // Device ID with standard values for i8042 mice
+            // BUS_I8042 (0x11): PS/2 interface bus type
+            // vendor (0x0002): Generic vendor ID for standard mice
+            // product (0x0001): Generic product ID for standard mice
+            // version (0x0001): Version 1.0 - standard PS/2 mouse protocol
+            id: InputId::new(InputId::BUS_I8042, 0x0002, 0x0001, 0x0001),
+
+            capability,
+        }
+    }
+}
+
+impl InputDevice for I8042Mouse {
+    fn name(&self) -> &str {
+        &self.name
+    }
+
+    fn phys(&self) -> &str {
+        &self.phys
+    }
+
+    fn uniq(&self) -> &str {
+        &self.uniq
+    }
+
+    fn id(&self) -> InputId {
+        self.id
+    }
+
+    fn capability(&self) -> &InputCapability {
+        &self.capability
+    }
+}
+
+fn handle_mouse_input(_trap_frame: &TrapFrame) {
+    if !I8042_CONTROLLER.is_completed() {
+        return;
+    }
+
+    let Some(controller) = I8042_CONTROLLER.get() else {
+        return;
+    };
+    let Some(data) = controller.lock().receive_data() else {
+        return;
+    };
+
+    let mut packet_state = MOUSE_PACKET_STATE.lock();
+    if let Some(events) = packet_state.process_byte(data) {
+        if let Some(registered_device) = REGISTERED_DEVICE.get() {
+            registered_device.submit_events(&events);
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum MouseType {
+    Standard,
+    IntelliMouse,
+}
+
+impl MouseType {
+    fn from_device_id(device_id: u8) -> Option<Self> {
+        const DEVICE_ID_INTELLIMOUSE: u8 = 0x03;
+        const DEVICE_ID_INTELLIMOUSE_EXPLORER: u8 = 0x04;
+
+        match device_id {
+            DEVICE_ID_INTELLIMOUSE | DEVICE_ID_INTELLIMOUSE_EXPLORER => {
+                Some(MouseType::IntelliMouse)
+            }
+            _ => Some(MouseType::Standard),
```

Comment:
Why should we return `Standard` for all other device IDs? Shouldn't we return `None` for an unrecognized device ID and then return `DeviceUnknown` in `init`?